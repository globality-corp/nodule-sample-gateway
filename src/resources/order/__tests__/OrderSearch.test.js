import request from 'supertest';
import { Nodule } from '@globality/nodule-config';
import { mockResponse } from '@globality/nodule-openapi';
import createApp from '../../../app';

let app;


beforeEach(async () => {
    await Nodule.testing().fromObject(
        mockResponse('charmander', 'order.search', {
            count: 2,
            items: [
                {
                    id: '00000000-0000-0000-0000-000000000000',
                    customerId: '00000000-0000-0000-0000-111111111111',
                },
                {
                    id: '00000000-0000-0000-0000-000000000000',
                    customerId: '00000000-0000-0000-0000-111111111112',
                },
            ],
        }),
    ).load();

    app = createApp();
});


it('should retrieve a list of orders', async () => {
    const query = `
        query {
          order {
            items {
                id
                customerId
            }
          }
        }
    `;

    const result = await request(app).post(
        '/gql/graphql',
    ).send({
        query,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});


it('should retrieve all orders for customer id', async () => {
    const query = `
        query($customerId: ID) {
          order(customerId: $customerId) {
            items {
                id
                customerId
            }
          }
        }
    `;

    const variables = {
        customerId: '00000000-0000-0000-0000-111111111111',
    };

    const result = await request(app).post(
        '/gql/graphql',
    ).send({
        query,
        variables,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});
