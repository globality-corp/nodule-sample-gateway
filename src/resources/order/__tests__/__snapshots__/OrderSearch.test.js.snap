// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrieve a list of orders 1`] = `
Object {
  "data": Object {
    "order": Object {
      "items": Array [
        Object {
          "customerId": "00000000-0000-0000-0000-111111111111",
          "id": "00000000-0000-0000-0000-000000000000",
        },
        Object {
          "customerId": "00000000-0000-0000-0000-111111111112",
          "id": "00000000-0000-0000-0000-000000000000",
        },
      ],
    },
  },
}
`;

exports[`should retrieve all orders for customer id 1`] = `
Object {
  "data": Object {
    "order": Object {
      "items": Array [
        Object {
          "customerId": "00000000-0000-0000-0000-111111111111",
          "id": "00000000-0000-0000-0000-000000000000",
        },
        Object {
          "customerId": "00000000-0000-0000-0000-111111111112",
          "id": "00000000-0000-0000-0000-000000000000",
        },
      ],
    },
  },
}
`;
